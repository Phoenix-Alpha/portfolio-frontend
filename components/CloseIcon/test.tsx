import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { CloseIcon } from '.'

describe('<CloseIcon />', () => {
  it('should renders correct <CloseIcon />', () => {
    const wrapper = shallow(<CloseIcon />)

    expect(wrapper.find('path').prop('stroke')).toMatch('#1f2837')
    expect(wrapper.find('svg').prop('width')).toEqual(16)
    expect(wrapper.find('svg').prop('height')).toEqual(16)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has correct width from width prop', () => {
    const width = 20

    const wrapper = shallow(<CloseIcon width={width} />)

    expect(wrapper.find('svg').prop('width')).toEqual(width)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has correct width from height prop', () => {
    const height = 20

    const wrapper = shallow(<CloseIcon height={height} />)

    expect(wrapper.find('svg').prop('height')).toEqual(height)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has correct color', () => {
    const colorHex = '#000'

    const wrapper = shallow(<CloseIcon colorHex={colorHex} />)

    expect(wrapper.find('path').prop('stroke')).toMatch(colorHex)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has correct classname', () => {
    const className = 'test-classname'

    const wrapper = shallow(<CloseIcon className={className} />)

    expect(wrapper.prop('className')).toContain(className)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
