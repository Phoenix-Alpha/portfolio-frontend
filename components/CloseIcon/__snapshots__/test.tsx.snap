// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseIcon /> should has correct classname 1`] = `
<svg
  aria-hidden="true"
  className="test-classname"
  fill="none"
  height={16}
  viewBox="0 0 16 16"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M14.4 1.6001L1.59998 14.4001M14.4 14.4001L1.59998 1.6001"
    stroke="#1f2837"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`<CloseIcon /> should has correct color 1`] = `
<svg
  aria-hidden="true"
  className=""
  fill="none"
  height={16}
  viewBox="0 0 16 16"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M14.4 1.6001L1.59998 14.4001M14.4 14.4001L1.59998 1.6001"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`<CloseIcon /> should has correct width from height prop 1`] = `
<svg
  aria-hidden="true"
  className=""
  fill="none"
  height={20}
  viewBox="0 0 16 16"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M14.4 1.6001L1.59998 14.4001M14.4 14.4001L1.59998 1.6001"
    stroke="#1f2837"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`<CloseIcon /> should has correct width from width prop 1`] = `
<svg
  aria-hidden="true"
  className=""
  fill="none"
  height={16}
  viewBox="0 0 16 16"
  width={20}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M14.4 1.6001L1.59998 14.4001M14.4 14.4001L1.59998 1.6001"
    stroke="#1f2837"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;

exports[`<CloseIcon /> should renders correct <CloseIcon /> 1`] = `
<svg
  aria-hidden="true"
  className=""
  fill="none"
  height={16}
  viewBox="0 0 16 16"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M14.4 1.6001L1.59998 14.4001M14.4 14.4001L1.59998 1.6001"
    stroke="#1f2837"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;
